ui/apps/platform/src/Containers/Policies/Modal/PolicyImport.utils.ts

```diff
@@ -1,31 +1,46 @@
+import { Policy } from 'types/policy.proto';
+import { ImportPolicyResponse } from 'services/PoliciesService';
+
 export const MIN_POLICY_NAME_LENGTH = 5;
 
-export const POLICY_DUPE_ACTIONS = {
-    KEEP_BOTH: 'keepBoth',
-    RENAME: 'rename',
-    OVERWRITE: 'overwrite',
-};
+type PolicyImportError = {
+    type: string;
+    duplicateName: string;
+    validationError?: string;
+    message: string;
+}
+
+export type PolicyImportDuplicateError = {
+    type: string;
+    incomingName?: string;
+    incomingId?: string;
+    duplicateName?: string;
+    validationError?: string | null;
+    message: string;
+}
 
 /**
  * parsePolicyImportErrors extracts any errors from the array of policies in the import, for ease-of-use in the UI
- *
- * @param   {array}  responses  a list of objects { succeeded: boolean, policy: object, errors: Array<object> }
- *
- * @return  {[array]}           Array< Array<{ type: string, incomingName: string, incomingId: string. duplicateName: string } > >
  */
-export function parsePolicyImportErrors(responses = []) {
-    const errors = responses.reduce((acc, res) => {
-        if (res?.errors?.length) {
-            const errorItems = res.errors.reduce((errList, err) => {
+export function parsePolicyImportErrors(responses: ImportPolicyResponse[]): PolicyImportDuplicateError[][] {
+    const errors = responses.reduce((acc: PolicyImportDuplicateError[][], res) => {
+        if (res.errors?.length) {
+            const errorItems = res.errors.reduce((errList:PolicyImportDuplicateError[] , err) => {
                 const thisErr = {
                     type: err.type,
-                    incomingName: res?.policy?.name,
-                    incomingId: res?.policy?.id,
-                    duplicateName: err.duplicateName,
-                    validationError: err?.validationError || null,
+                    incomingName: res.policy?.name,
+                    incomingId: res.policy?.id,
                     message: err.message,
                 };
 
+                if (err.type === 'duplicate_id' || err.type === 'duplicate_name') {
+                    thisErr['duplicateName'] = err.duplicateName
+                }
+
+                if (err.type === 'invalid_policy') {
+                    thisErr['validationError'] =  err.validationError || null
+                }
```

Comment:
What's the reason for wrapping these in conditionals?

Reply:
ts was complaining that the types didn't line up since `duplicateName` and `validationError` were optional based on the type of error it was. i'll try mark's typing suggestion below to see if it fixes the problem